maitake/src/task.rs

```diff
@@ -161,14 +179,28 @@ pub(crate) struct Header {
 }
 
 enum Cell<F: Future> {
-    Future(F),
-    Finished(F::Output),
+    /// The future is still pending.
+    Pending(F),
+    /// The future has completed, and its output is ready to be taken by a
+    /// `JoinHandle`, if one exists.
+    Ready(F::Output),
+    /// The future has completed, and the task's output has been taken or is not
+    /// needed.
+    Joined,
 }
 
-#[derive(Debug)]
+// #[derive(Debug)]
```

Comment:
Remove or restore?

Reply:
oh, i had to add a manual impl of this; i should remove the commented out derive attribute. good catch!